moto/iam/urls.py

```diff
@@ -1,6 +1,6 @@
 from __future__ import unicode_literals
 from .responses import IamResponse
 
-url_bases = ["https?://iam(.*).amazonaws.com"]
+url_bases = [r"https?://iam\.(.*)amazonaws\.com"]
```

Comment:
`url_bases = [r"https?://iam\.(.*\.)?amazonaws\.com"]`

if we want to include:
https://iam.amazonaws.com
https://iam.anything.amazonaws.com (in which case is it necessary?)

Reply:
IAM is a global service, so boto3 sends everything to `iam.amazonaws.com` AFAIK. I'll leave the URL open for other regions though (`iam.us-east-1.amazonaws.com`), just in case other SDK's behave differently.